src/site/content/en/angular/route-preloading/index.md

```diff
@@ -0,0 +1,126 @@
+  - mgechev
+---
+
+Route-level code-splitting can help you reduce the initial load time of an application by delaying the JavaScript associated with routes that aren't initially needed. This way, the Angular router waits until a user navigates to a given route before triggering a network request to download the associated JavaScript.
+
+While this technique is great for initial page load, it can slow down navigation, depending on the users' network latency and bandwidth. One way to tackle this problem is **route preloading**. Using preloading, when the user is at a given route, you can download and cache JavaScript associated with routes that are likely to be needed next. The Angular router provides this functionality out of the box.
+
+In this post, you'll learn how to speed up navigation when using route-level code-splitting by taking advantage of JavaScript preloading in Angular.
+
+## Route preloading strategies in Angular
+
+The Angular router provides a configuration property called `preloadingStrategy`, which defines the logic for preloading and processing lazy-loaded Angular modules. We'll cover two possible strategies:
+
+* `PreloadAllModules`, which preloads all lazy-loaded routes, as the name implies
+* `QuicklinkStrategy`, which preloads only the routes associated with links on the current page.
+
+<p style="text-align: right">
+<em>The rest of this post refers to a sample Angular app. You can find the source code on <a href="https://github.com/mgechev/route-preloading-web-dev">GitHub</a>.</em></p>
+
+### Using the `PreloadAllModules` strategy
+
+The sample app has several lazy-loaded routes. To preload all of them using the `PreloadAllModules` strategy—which is built into Angular—specify it as the value for the `preloadingStrategy` property in the router configuration:
+
+```js
+import { RouterModule, PreloadAllModules } from '@angular/router';
+// ...
+
+RouterModule.forRoot([
+  ...
+], {
+  preloadingStrategy: PreloadAllModules
+})
+// ...
+```
+
+Now let's serve the application and look at the **Network** panel in Chrome DevTools:
+
+{% Instruction 'devtools-network', 'ol' %}
+
+You should see that the router downloaded `nyan-nyan-module.js` and `about-about-module.js` in the background when you opened the application:
+
+<figure class="w-figure w-figure--fullbleed">
+<video controls loop muted poster="https://storage.googleapis.com/web-dev-angular/preloading/poster.png">
+  <source src="https://storage.googleapis.com/web-dev-angular/preloading/preload-all.webm" type="video/webm; codecs=vp8">
+  <source src="https://storage.googleapis.com/web-dev-angular/preloading/preload-all.mp4" type="video/mp4; codecs=h264">
+</video>
+ <figcaption class="w-figcaption w-figcaption--fullbleed">
+    Fig. 1 — PreloadAll Angular strategy
+  </figcaption>
+</figure>
+
+The router also registered the modules' route declarations so that when you navigate to a URL associated with any of the preloaded modules, the transition is instantaneous.
```

Comment:
This is actually the idea for which I thought a video would be most helpful: it's hard to see that the transitions are instantaneous without a video. What do you think?

Reply:
I wanted to show the timing of prefetching the bundles. Maybe I can provide one more video showing slow navigation and another showing faster navigation after prefetching.

Reply:
That actually sounds kind of onerous. Let's give it a pass, and I'll approve the post :)